psycopg/psycopg/types/numpy.py

```diff
@@ -0,0 +1,186 @@
+"""
+Adapters for numpy types.
+"""
+
+# Copyright (C) 2022 The Psycopg Team
+
+from typing import Any, Dict, Optional
+
+from psycopg.types.numeric import dump_int_to_numeric_binary
+
+import numpy as np
+
+from .. import postgres
+from .._struct import pack_int2, pack_int4, pack_int8, pack_float4, pack_float8
+from ..abc import AdaptContext, Buffer
+from ..adapt import Dumper
+from ..pq import Format
+
+
+class _NPIntDumper(Dumper):
+    def dump(self, obj: Any) -> bytes:
+        return str(obj).encode()
+
+    def quote(self, obj: Any) -> bytes:
+        value = self.dump(obj)
+        return value if obj >= 0 else b" " + value
+
+
+class NPInt2Dumper(_NPIntDumper):
+    oid = postgres.types["int2"].oid
+
+
+class NPInt4Dumper(_NPIntDumper):
+    oid = postgres.types["int4"].oid
+
+
+class NPInt8Dumper(_NPIntDumper):
+    oid = postgres.types["int8"].oid
+
+
+class NPBooleanDumper(_NPIntDumper):
+    oid = postgres.types["bool"].oid
+
+    def dump(self, obj: np.bool_) -> bytes:
+        return "t".encode() if bool(obj) is True else "f".encode()
+
+
+class NPIntNumericDumper(_NPIntDumper):
+    oid = postgres.types["numeric"].oid
+
+
+class _SpecialValuesDumper(Dumper):
+
+    _special: Dict[bytes, bytes] = {}
+
+    def dump(self, obj: Any) -> Buffer:
+        return str(obj).encode()
+
+    def quote(self, obj: Any) -> Buffer:
+        value = self.dump(obj)
+
+        if value in self._special:
+            return self._special[value]
+
+        return value if obj >= 0 else b" " + value
+
+
+class NPFloatDumper(_SpecialValuesDumper):
+
+    oid = postgres.types["float8"].oid
+
+    _special = {
+        b"inf": b"'Infinity'::float8",
+        b"-inf": b"'-Infinity'::float8",
+        b"nan": b"'NaN'::float8",
+    }
+
+
+class NPFloat4Dumper(NPFloatDumper):
```

Comment:
Nit: the general convention in the project is to give dumpers a name similar to the python object they dump, and to loaders a name similar to the Postgres type they load.

Am I too pedantic if I ask to call the dumpers with a name similar to the numpy type name?

```suggestion
class NPFloat32Dumper(NPFloatDumper):
```